packages/build/src/plugins_core/report_functions.ts

```diff
@@ -0,0 +1,42 @@
+import { promises as fs } from 'fs'
+import { join, resolve } from 'path'
+
+import { closeClient, formatTags, InputStatsDOptions, startClient, validateStatsDOptions } from '../report/statsd.js'
+
+const DEFAULT_EDGE_FUNCTIONS_DIST = '.netlify/edge-functions-dist/'
+
+/**
+ * Record number of functions build and differentiate between autogenerated and user generated.
+ * Sends to statsd daemon.
+ */
+export const reportFunctions = async function (statsdOpts: InputStatsDOptions, netlifyConfig): Promise<void> {
+  if (!validateStatsDOptions(statsdOpts)) {
+    return
+  }
+
+  const buildDir = netlifyConfig?.build?.publish
+  const edgeFunctionsDistPath = resolve(buildDir, DEFAULT_EDGE_FUNCTIONS_DIST)
+  const manifestPath = join(edgeFunctionsDistPath, 'manifest.json')
+  try {
+    await fs.access(manifestPath)
+
+    const data = await fs.readFile(manifestPath)
```

Comment:
We get access to the manifest [here](https://github.com/netlify/build/blob/main/packages/build/src/plugins_core/edge_functions/index.ts#L58), so it feels a bit odd to be reading that file from disk and parsing that data again in another place.

I understand that these things happen in different parts of the codebase, but even if this a necessary evil for now, can we work towards (and track the effort required to) a more sustainable implementation?

I can see this type of thing being useful in other core steps — how would we handle it if we weren't persisting the manifest to disk in the first place? Feels like allowing core steps to return these metrics (via an object, a callback, or something else) would be a useful building block.

Reply:
Makes sense! I'm going to do some changes :)